VocaDbModel/Service/OtherService.cs

```diff
@@ -275,48 +264,16 @@ private IEnumerable<EntryWithCommentsContract> CreateEntryWithCommentsContract<T
 			return comments.GroupBy(e => e.Entry, new EntryComparer()).Select(e => new EntryWithCommentsContract(entryContractFac(e.First()), e.Select(c => new CommentContract(c)).ToArray()));
 		}
 
-		private async Task<List<Comment>> GetComments<TEntry, TComment>(ISession session, int maxComments, bool checkDeleted) where TComment : GenericComment<TEntry> where TEntry : class, IEntryWithNames
-		{
-			var q = session.Query<TComment>();
-
-			if (checkDeleted)
-				q = q.Where(c => !c.EntryForComment.Deleted);
-
-			return await q.OrderByDescending(c => c.Created).Take(maxComments).Cast<Comment>().VdbToListAsync();
-		}
-
 		private async Task<EntryWithCommentsContract[]> GetRecentCommentsAsync(ISession session, int maxComments)
 		{
-			var albumComments = await GetComments<Album, AlbumComment>(session, maxComments, true);
-			var artistComments = await GetComments<Artist, ArtistComment>(session, maxComments, true);
-			var songComments = await GetComments<Song, SongComment>(session, maxComments, true);
-			var discussionComments = await GetComments<DiscussionTopic, DiscussionComment>(session, maxComments, true);
-			var songListComments = await GetComments<SongList, SongListComment>(session, maxComments, false);
-			var tagComments = await GetComments<Tag, TagComment>(session, maxComments, true);
-			var eventComments = await GetComments<ReleaseEvent, ReleaseEventComment>(session, maxComments, true);
-
-			// Discussion topics aren't actually comments but we want to show them in the recent comments list anyway
-			var discussionTopics = await session.Query<DiscussionTopic>().Where(c => !c.Deleted).OrderByDescending(c => c.Created).Take(maxComments).VdbToListAsync();
-			var discussionTopicsAsComments = discussionTopics.Select(t => new DiscussionComment(t, t.Content, new AgentLoginData(t.Author, t.AuthorName ?? t.Author.Name))
-			{
-				Created = t.Created
-			});
-
-			var albumReviews = await session.Query<AlbumReview>().OrderByDescending(r => r.Date).Take(maxComments).VdbToListAsync();
-			var albumReviewsAsComments = albumReviews.Select(r => new AlbumComment(r.Album, r.Text, new AgentLoginData(r.User))
-			{
-				Created = r.Date
-			});
+			var comments = (await session.Query<Comment>()

```

Comment:
Since you're doing the ```.Where(c => !c.Entry.Deleted);``` filtering last after ToList, isn't it going to return less comments if there are deleted entries?

Reply:
Yes, but this happens in activity entries as well. I don't think we can directly query `Entry.Deleted` before `ToList`. (an ORM limitation?) I think we could have a column something like `EntryDeleted` as a workaround.

Reply:
Right. I didn't notice Entry is an accessor property (it's hard when doing review on GitHub). ORM can't map those directly. Maybe there is a workaround since this is a common situation (accessing polymorphic property). For now it's ok like this too, but maybe add a comment?